pkg/cli/cmd.go

```diff
@@ -84,6 +86,35 @@ func initializeConfig(cmd *cobra.Command) error {
 	return nil
 }
 
+func defaultForMissingFlagPossible(flag, defaultKey string) bool {
```

Comment:
I see you chose to implement a global mechanism for using default project and codeset values if they're not explicitly supplied as arguments where they're needed for their respective sub-commands. This is one way to do it, but now you have two places where you need to keep flags in sync for applicable sub-commands: here and where the cobra command is defined.

I would have spread it out across all commands where this is applicable instead of in a central location. That way, it's more obvious where you need this particular type of behavior. I'm not saying you should duplicate this code everywhere you have a codeset or project argument, but it should be something that you can configure for each cobra command explicitly, preferably in the same area of code where you define the flags.

Here's a crazy idea: what if you're "linking" all those individual flags to the same global variables ? For example, instead of linking the `workflow assign {-p|--codeset-project}` flag to the local `assignOptions.codesetProject` variable, link it to the global one that is also linked to the currently set project. That way, you only have one variable you need to care about for all flags that point to a project _and_ you get some indication in each cobra command that this is the behavior you want. It might take some hacking to make it work with the concept of a mandatory flag, but I think that's doable.

Reply:
OK, maybe I'm missing something but this seems like I should treat Project/Codeset as global options, like e.g. `url` option and related `FUSEML_SERVER_URL` env var.

Then it would work without that ugly `defaultForMissingFlagPossible` processing, however, it would reveal global Project/Codeset options to all commands, even those where they are not needed.

If, on the other hand, I would completely ignore it on the global level and only try to handle it on the specific subcommand level, I would have a problem (as you hinted) with those flags marked as mandatory because the check for mandatory flags is processed before parsing the config file ....

_but it should be something that you can configure for each cobra command explicitly, preferably in the same area of code where you define the flags._

Even if the problem with mandatory flags could be avoided, I'm not sure about this part. This would require extra config file processing for each subcommand, apart from the one that is done globaly

Reply:
Let me phrase it yet differently: I'm quite sure it could be done with other approach than the one I chose and that it could be better in some aspects. 
However I have a feeling it would mean just some different 'hackish' (read: ugly) solutions in different parts of the code then the ones I modified.

I could be wrong of course, it took me some time to understand how this cobra/viper combination works and I could have missed something obvious.